erpnext/__init__.py

```
@@ -1,195 +1,184 @@
+            if isinstance(ctx, expected_type):
+                pass
+            elif isinstance(ctx, Document):
+                ctx = expected_type(**ctx.as_dict())
+            elif isinstance(ctx, dict):
+                ctx = expected_type(**ctx)
+            elif isinstance(ctx, str):
+                ctx = expected_type(**frappe.parse_json(ctx))
+            else:
+                raise TypeError(
+                    f"Unsupported ctx type: {type(ctx)!r}; expected {expected_type}, Document, dict, or JSON string"
+                )
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)